src/misc/support.cpp

```diff
@@ -392,3 +392,25 @@ const std_fs::path &GetExecutablePath()
 	}
 	return exe_path;
 }
+
+const Bit8u DOS_DATE_months[] = {0,  31, 28, 31, 30, 31, 30,
+                                 31, 31, 30, 31, 30, 31};
+
+bool is_date_valid(uint32_t year, uint32_t month, uint32_t day)
+{
+	if (year < 1980 || month > 12 || month == 0 || day == 0)
+		return false;
+	else if (month == 2 && day == 29 &&
+	         (year % 4 != 0 || (year % 100 == 0 && year % 400 != 0)))
```

Comment:
This make's sure we consider February's 29th day invalid if we're in a normal (non-leap) year.
So this handles the non-leap-year scenario.

In a leap-year scenario, this block will always be false: `(year % 4 != 0 || (year % 100 == 0 && year % 400 != 0))`, so this chunk will always pass-through.

This means that the next statement: `(month != 2 && day > DOS_DATE_months[month])` needs to cover the leap-year scenario.

So I can see if we use:
 - year: 2024
 - month: 02
 - day: > 29, say 200

we will get a positive return.

Reply:
Thanks for the analysis. I have revised the code as suggested. Thanks!